carapace-server/src/main/java/org/carapaceproxy/server/HttpProxyServer.java

```diff
@@ -553,6 +554,22 @@ public RuntimeServerConfiguration buildValidConfiguration(ConfigurationStore sim
         return newConfiguration;
     }
 
+    public void createDynamicCertificateForDomain(CertificateData cert) throws Exception {
+        // Certificate saving on db
+        dynamicConfigurationStore.saveCertificate(cert);
+
+        // Configuration updating with new certificate (whether not yet existent)
+        Properties props = dynamicConfigurationStore.asProperties(null);
+        if (!dynamicConfigurationStore.anyPropertyMatches("certificate\\.[0-9]+\\.hostname", cert.getDomain())) {
```

Comment:
cert.getDomain() -> this is not a regexp
I have suggested to use BiPredicate or BiFunction<Boolean, String, String> this way the match login will be here and not pushed into the ConfigurationStore